kong/constants.lua

```diff
@@ -42,24 +48,39 @@ for _, plugin in ipairs(deprecated_plugins) do
   deprecated_plugin_map[plugin] = true
 end
 
+local headers = {
+  HOST_OVERRIDE = "X-Host-Override",
+  PROXY_LATENCY = "X-Kong-Proxy-Latency",
+  UPSTREAM_LATENCY = "X-Kong-Upstream-Latency",
+  CONSUMER_ID = "X-Consumer-ID",
+  CONSUMER_CUSTOM_ID = "X-Consumer-Custom-ID",
+  CONSUMER_USERNAME = "X-Consumer-Username",
+  CREDENTIAL_USERNAME = "X-Credential-Username",
+  RATELIMIT_LIMIT = "X-RateLimit-Limit",
+  RATELIMIT_REMAINING = "X-RateLimit-Remaining",
+  CONSUMER_GROUPS = "X-Consumer-Groups",
+  FORWARDED_HOST = "X-Forwarded-Host",
+  FORWARDED_PREFIX = "X-Forwarded-Prefix",
+  ANONYMOUS = "X-Anonymous-Consumer",
+  VIA = "Via",
+  SERVER = "Server"
+}
+
 return {
   PLUGINS_AVAILABLE = plugin_map,
   DEPRECATED_PLUGINS = deprecated_plugin_map,
   -- non-standard headers, specific to Kong
-  HEADERS = {
-    HOST_OVERRIDE = "X-Host-Override",
-    PROXY_LATENCY = "X-Kong-Proxy-Latency",
-    UPSTREAM_LATENCY = "X-Kong-Upstream-Latency",
-    CONSUMER_ID = "X-Consumer-ID",
-    CONSUMER_CUSTOM_ID = "X-Consumer-Custom-ID",
-    CONSUMER_USERNAME = "X-Consumer-Username",
-    CREDENTIAL_USERNAME = "X-Credential-Username",
-    RATELIMIT_LIMIT = "X-RateLimit-Limit",
-    RATELIMIT_REMAINING = "X-RateLimit-Remaining",
-    CONSUMER_GROUPS = "X-Consumer-Groups",
-    FORWARDED_HOST = "X-Forwarded-Host",
-    FORWARDED_PREFIX = "X-Forwarded-Prefix",
-    ANONYMOUS = "X-Anonymous-Consumer"
+  HEADERS = headers,
+  -- TODO figure out how to organize headers and bit positions in code
+  -- another (better?) way could be to keep the mask together 
+  -- with header definitions, but not all headers have poitions..
+  HEADER_MASKS = {
+    [headers.PROXY_LATENCY] = 0x01,
+    [headers.UPSTREAM_LATENCY] = 0x02,
+    [headers.SERVER] = 0x04,
+    [headers.VIA] = 0x08,
+    latency = bor(0x01,0x02),
+    server = bor(0x04,0x08),
```

Comment:
This one could be `server_tokens` (and the property in the configuration could be `headers` probably?) Sorry if I mislead you while drafting the quick implementation on the fly while we chatted. What do you think?

Reply:
I put in names just as placeholders since I was having a difficult time in deciding on which one to chose.

`headers` as a property in user visible configuration property is much easier to grok.
I understand that `server_tokens` is coming from NGINX world and we would like to stick to the conventions but could that instead be `server_names`? Is it a bit more descriptive?
I'm thinking:
```
headers = server_names, kong_latencies, X-Kong-Foo-Bar
```